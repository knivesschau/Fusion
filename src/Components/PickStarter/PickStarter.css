.Starter_Recipes {
    text-align: center;
    border: 1px solid black;
    margin: 2em;
}

#starter-recipes {
    text-align: left;
    padding-left: 1em;
}