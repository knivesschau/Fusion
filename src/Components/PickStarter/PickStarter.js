import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import fusionContext from '../../fusionContext';
import './PickStarter.css';

export default class PickStarter extends Component {
    
    static contextType = fusionContext; 

    render() {
        
        const {bases=[]} = this.context;

        return (
            <>
            <section className="Pick_Starter">

                <div className="Starter_Recipes">
                    <h1 id="begin-fuse">Begin Fusing</h1>
                    <h2 id="select-recipe">Select a Starter Recipe to Begin Fusing:</h2>
                    
                    <ul id="starter-recipes">
                    {bases.map((base, i) => {
                        return (
                            <li key={i}>
                                <Link to={`/bases/${base.recipe_id}`} key={i}>
                                    <h3>{base.base_name} ({base.cuisine_name})</h3>
                                </Link>
                            </li>
                        );
                    })}
                    </ul>
                </div>
            </section>  
            </>
        );
    };
};