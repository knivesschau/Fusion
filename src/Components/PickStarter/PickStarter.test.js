import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import PickStarter from './PickStarter';

describe ('PickStarter Component', () => {
    it ('Renders the complete PickStarter Component and all nested Components', () => {
        const wrapper = shallow(<PickStarter/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});