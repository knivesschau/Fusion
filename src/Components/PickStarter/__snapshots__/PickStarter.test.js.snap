// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickStarter Component Renders the complete PickStarter Component and all nested Components 1`] = `
<Fragment>
  <section
    className="Pick_Starter"
  >
    <div
      className="Starter_Recipes"
    >
      <h1
        id="begin-fuse"
      >
        Begin Fusing
      </h1>
      <h2
        id="select-recipe"
      >
        Select a Starter Recipe to Begin Fusing:
      </h2>
      <ul
        id="starter-recipes"
      />
    </div>
  </section>
</Fragment>
`;
